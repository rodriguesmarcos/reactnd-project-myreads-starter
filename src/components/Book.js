import React from 'react';
import PropTypes from 'prop-types';
import BookShelfChanger from './BookShelfChanger';

const Book = ({book, shelves, onChangeShelf}) => (
  <div className="book">

    <input type="checkbox" name="bookId" value={book.id}/>

    <div className="book-top">
      <div className="book-cover" style={{
        width: 128,
        height: 193,
        backgroundImage: `url(${book.imageLinks.smallThumbnail && (book.imageLinks.smallThumbnail)})` }}></div>
      <div className="book-shelf-changer">
        <BookShelfChanger
          book={book}
          shelves={shelves}
          onChangeShelf={onChangeShelf}
        />
      </div>
    </div>
    <div className="book-title">{book.title}</div>
    <div className="book-authors">{Array.isArray(book.authors) && (book.authors.join(', '))}</div>
  </div>
);

Book.propTypes = {
  book: PropTypes.shape({ id: PropTypes.string.isRequired }).isRequired,
  shelves: PropTypes.array.isRequired,
  onChangeShelf: PropTypes.func.isRequired,
}

export default Book;
