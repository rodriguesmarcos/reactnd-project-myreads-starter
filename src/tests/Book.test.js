import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Book from '../components/Book.js';
import shelves from './fixtures/shelves';
import books from './fixtures/books';

describe('<Book />', () => {

  const props = {
    book: books[0],
    shelves,
    onChangeShelf: jest.fn(),
  }

  test('renders without exploding', () => {
    const wrapper = shallow(<Book {...props}/>);
    expect(wrapper.length).toBe(1);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test( 'should have a checkbox', () => {
    const wrapper = shallow(<Book {...props}/>);
    const checkbox = wrapper.find('input[type="checkbox"]');
    expect(checkbox.length).toBe(1);
    expect(checkbox.prop('value')).toBe(`${books[0].id}`);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test( 'should have a Link to BookDetails', () => {
    const wrapper = shallow(<Book {...props}/>);
    const checkbox = wrapper.find('Link');
    expect(checkbox.length).toBe(1);
    expect(checkbox.prop('to')).toBe(`/book/${books[0].id}`);
    expect(toJson(wrapper)).toMatchSnapshot();
  });


});
