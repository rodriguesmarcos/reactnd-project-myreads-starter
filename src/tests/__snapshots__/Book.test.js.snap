exports[`<Book /> renders without exploding 1`] = `
<div
  className="book">
  <input
    name="bookId"
    type="checkbox"
    value="nggnmAEACAAJ" />
  <Link
    className="book-link"
    replace={false}
    to="/book/nggnmAEACAAJ">
    <div
      className="book-top">
      <div
        className="book-cover"
        style={
          Object {
            "backgroundImage": "url(http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=5&source=gbs_api)",
            "height": 193,
            "width": 128,
          }
        } />
    </div>
    <div
      className="book-title">
      The Linux Command Line
    </div>
    <div
      className="book-authors">
      William E. Shotts, Jr.
    </div>
  </Link>
  <div
    className="book-shelf-changer">
    <Route
      render={[Function]} />
  </div>
</div>
`;

exports[`<Book /> should have a Link to BookDetails 1`] = `
<div
  className="book">
  <input
    name="bookId"
    type="checkbox"
    value="nggnmAEACAAJ" />
  <Link
    className="book-link"
    replace={false}
    to="/book/nggnmAEACAAJ">
    <div
      className="book-top">
      <div
        className="book-cover"
        style={
          Object {
            "backgroundImage": "url(http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=5&source=gbs_api)",
            "height": 193,
            "width": 128,
          }
        } />
    </div>
    <div
      className="book-title">
      The Linux Command Line
    </div>
    <div
      className="book-authors">
      William E. Shotts, Jr.
    </div>
  </Link>
  <div
    className="book-shelf-changer">
    <Route
      render={[Function]} />
  </div>
</div>
`;

exports[`<Book /> should have a checkbox 1`] = `
<div
  className="book">
  <input
    name="bookId"
    type="checkbox"
    value="nggnmAEACAAJ" />
  <Link
    className="book-link"
    replace={false}
    to="/book/nggnmAEACAAJ">
    <div
      className="book-top">
      <div
        className="book-cover"
        style={
          Object {
            "backgroundImage": "url(http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=5&source=gbs_api)",
            "height": 193,
            "width": 128,
          }
        } />
    </div>
    <div
      className="book-title">
      The Linux Command Line
    </div>
    <div
      className="book-authors">
      William E. Shotts, Jr.
    </div>
  </Link>
  <div
    className="book-shelf-changer">
    <Route
      render={[Function]} />
  </div>
</div>
`;
